sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs

```diff
@@ -141,6 +146,21 @@ public void Dispose()
 
             foreach (ProducedDiagnosticScope producedDiagnosticScope in Scopes)
             {
+                var activity = producedDiagnosticScope.Activity;
+                var operationName = activity.OperationName;
+                // traverse the activities and check for duplicates among ancestors
+                while (activity != null)
+                {
+                    if (operationName == activity.Parent?.OperationName)
```

Comment:
Any chance `operationName` could be null here and cause a false positive?

Reply:
I don't think so, but if it was null there is something wrong so it wouldn't be a false-positive 😄